CosmicsAnalyzer/EarthAsDMAnalyzer/plugins/EarthAsDMAnalyzer.cc

```diff
@@ -293,8 +399,96 @@ void EarthAsDMAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup&
     }
     
     muon_n_++ ;
+  } // end of muon collection loop
+
+  if (!isData_ && closestRecIndex >= 0) {
+    float genPt = (*genColl)[1].pt();
+    float genPhi = (*genColl)[1].phi();
+    reco::MuonRef matchedMuon  = reco::MuonRef( muonCollectionHandle, closestRecIndex );
+
+    if (verbose_ > 2) 
+        LogPrint(MOD) << "  >> dRMinGen:" << dRMinGen << ", closestRecIndex:" << closestRecIndex
+                      << ", GEN muon pT: " << genPt
+                      << ", Reco muon pT: "   << matchedMuon->pt();
+    
+    // 1D plot of reco and gen pt, phi
+    GenPt->Fill(genPt);
+    GenPhi->Fill(genPhi);
+    RecoPt->Fill(matchedMuon->pt());
+    RecoPhi->Fill(matchedMuon->phi());
+    RelDiffTrackPtAndTruthPt->Fill( (matchedMuon->pt() - genPt) / genPt );
+    RelDiffTrackPhiAndTruthPhi->Fill( (matchedMuon->phi() - genPhi) / genPhi );
```

Comment:
maybe these should be saved in the ntuple too (and not use the histos here, but rather when reading the ntuple in like we do with the rest of the variable)